### PR TITLE
Add ERC: Opaque Token

### PR DESCRIPTION
**Pull request by ivica7**
Opaque Token - initial ERC draft

### CODE REVIEW

File: erc-draft_opaque_token.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
+---
```
Comment (abcoathup) [inline erc-draft_opaque_token.md:1]:
```suggestion
---
eip: 7722
```
Assigning next sequential EIP/ERC/RIP number.

Please also update the filename.

File: erc-draft_opaque_token.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+title: Opaque Token
+description: An opaque token standard with privacy support
+author: Ivica Aračić (@ivica7), SWIAT
+discussions-to: https://ethereum-magicians.org/t/erc-idea-opaque-token-a-token-standard-with-privacy-support-that-unifies-fungible-and-non-fungible-tokens
```
Comment (abcoathup) [inline erc-draft_opaque_token.md:5]:
```suggestion
discussions-to: https://ethereum-magicians.org/t/erc-7722-opaque-token/20249
```

File: ERCS/erc-7722.md [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+created: 2024-06-09
+---
+
+## Abstract
+
+This ERC proposes a specification for an opaque token that enhances privacy by concealing balance information. Privacy is achieved by representing balances as off-chain data encapsulated in hashes, referred to as "baskets". These baskets can be reorganized, transferred, and managed through token functions on-chain.
+
+## Motivation
+
+In [ERC-20](./eip-20.html), token balances are transparent on-chain. While some privacy can be achieved by using pseudonyms for the sender and receiver, the risk of revealing the entire portfolio to the public increases when using smart contract accounts like Gnosis Safe or [ERC-725](./eip-725.html). This is particularly problematic in regulated environments where KYC (Know Your Customer) and KYBP (Know Your Business Partner) procedures are required.
```
Comment (SamWilsn) [inline ERCS/erc-7722.md:19]: Linking to ERC-725 creates an implicit dependency, and so your proposal will not be able to advance past ERC-725's status (currently Draft.) This isn't a problem; just wanted to let you know.

Comment (SamWilsn) [inline ERCS/erc-7722.md:19]: It is unclear from the text why using a smart contract account has a higher risk of revealing the entire portfolio.

File: ERCS/erc-7722.md [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+
+This proposal aims to conceal balances on-chain, allowing the use of smart contract accounts to hold tokens without compromising privacy or integrity.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
```
Comment (SamWilsn) [inline ERCS/erc-7722.md:26]: This section could do with an overview of the entire system. It's a bit hard to grok just from the interfaces and endpoints.

File: ERCS/erc-7722.md [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+We have chosen the token operator, which in most cases will be the issuer or registrar, as the initial and main source for off-chain data. This is acceptable, since they must know anyway which investor holds which positions to manage lifecycle events on the token. While this approach may not be suitable for every use case within the broader Ethereum ecosystem, it fits well the financial instruments in the regulated environment of the financial industry, which rely on strict KYC and token operation procedures.
+
+## Backwards Compatibility
+
+* Opaque Token is not compatible with ERC-20 for reasons explained in the Rationale section.
+
+## Reference Implementation
+
+To be included as inline code or in `../assets/eip-####/`.
+
```
Comment (SamWilsn) [inline ERCS/erc-7722.md:314]:
```suggestion
```

Can add this section when you're ready.

File: ERCS/erc-7722.md [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+## Security Considerations
+
+### Fraudulent Oracles
+To be discussed...
```
Comment (SamWilsn) [inline ERCS/erc-7722.md:318]:
```suggestion
<!-- TODO -->
```

File: ERCS/erc-7722.md [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+
+### Oracles Collecting Confidential Data
+To be discussed...
```
Comment (SamWilsn) [inline ERCS/erc-7722.md:321]:
```suggestion
<!-- TODO -->
```

File: ERCS/erc-7722.md [hunk 8]
```diff
@@ -0,0 +1,328 @@
...
+
+### Confidential Data Loss
+To be discussed...
```
Comment (SamWilsn) [inline ERCS/erc-7722.md:324]:
```suggestion
<!-- TODO -->
```

The linter will make sure you replace HTML-style comments before you get too far in the process.

File: ERCS/erc-7722.md (hunk 3 above)
Comment (ivica7) [reply ERCS/erc-7722.md:19]: good to know! thanks for pointing out!

Comment (ivica7) [reply ERCS/erc-7722.md]: the section on motivation has been updated.

File: ERCS/erc-7722.md (hunk 4 above)
Comment (ivica7) [reply ERCS/erc-7722.md:26]: added a summary about the core concept around which this token proposal revolves around.

**Review by SamWilsn — APPROVED** [review]

**Pull request merged**